src/parser/Parser.ts

```diff
@@ -174,6 +174,17 @@ export class Parser {
             } else {
                 name = consume(`Expected ${functionType.text} name after '${functionType.text}'`, Lexeme.Identifier) as Identifier;
                 leftParen = consume(`Expected '(' after ${functionType.text} name`, Lexeme.LeftParen);
+
+                //prevent functions from ending with type designators
+                let lastChar = name.text[name.text.length - 1];
+                if (["$", "%", "!", "#"].indexOf(lastChar) > -1) {
+                    //Since function type is first word, force the first char to upper case to match style of other error messages
+                    let properFunctionName = functionType.text[0].toUpperCase() + functionType.text.substring(1);
```

Comment:
Would you mind reformatting the error message so the function name isn't the first word?  Something like `Function named  '${name.text}' cannot end with type designator '${lastChar}'` would let us keep the function name capitalized the way it's written in the source text &mdash; even if it's `rEaLlYsilLy$()`.  Then we wouldn't need this line!

Reply:
Done.

Reply:
Although, not sure if you noticed, this wasn't the function name. it was the function TYPE (literally the words `Sub` or `Function`)

Reply:
Oh heck, I got thrown off by the `properFunctionName` variable, thinking it was the function's name but formatted as a proper noun!  Sorry about that!